cmd/server/api/handler.go

```diff
@@ -0,0 +1,54 @@
+package api
+
+import (
+	"net/http"
+	"time"
+)
+
+// Define an interface for HTTP client operations
+type HttpClientInterface interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
+// Create a wrapper for the net/http client
+type HttpClientWrapper struct {
+	Client *http.Client
+}
+
+func (h *HttpClientWrapper) Do(req *http.Request) (*http.Response, error) {
+	return h.Client.Do(req)
+}
+
+type ApiHandlerInterface interface {
+	PostRpcHandler(w http.ResponseWriter, r *http.Request)
+}
+
+type ApiHandler struct {
+	RpcUrl     string
+	HttpClient HttpClientInterface // Inject the HTTP client interface here
+}
+
+type ApiHandlerOption func(*ApiHandler)
+
+func WithHttpClient(client HttpClientInterface) ApiHandlerOption {
+	return func(h *ApiHandler) {
+		h.HttpClient = client
+	}
+}
+
+func NewApiHandler(rpcUrl string, opts ...ApiHandlerOption) *ApiHandler {
```

Comment:
Similar to the previous comment. You should avoid repeating the package name for naming structs, funcs, interfaces... to make it idiomatic as the official Go guide explains it here: https://go.dev/doc/effective_go#names